src/tdastro/astro_utils/passbands.py

```diff
@@ -0,0 +1,249 @@
+import logging
+import os
+import socket
+import urllib.request
+from pathlib import Path
+from urllib.error import HTTPError, URLError
+
+import numpy as np
+import scipy.integrate
+
+from tdastro.sources.physical_model import PhysicalModel
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")
+
+
+class Passbands:
+    """Class to handle the transmission tables and normalization for different passbands.
+
+    Attributes
+    ----------
+    bands : list
+        A list of strings representing the band names.
+    data_path : str
+        Path to the directory where the bandpass data is stored.
+    transmission_tables : dict
+        A dictionary with band names as keys and transmission tables as values. Transmission tables
+        are 2D arrays with wavelengths (Angstrom) in the first column and transmission strengths in
+        the second.
+    normalized_system_response_tables : dict
+        A dictionary with band names as keys and normalized system response tables as values. These
+        tables are 2D arrays with wavelengths (Angstrom) in the first column and normalized system
+        response values in the second.
+    """
+
+    def __init__(self, bands=None):
+        if bands is None:
+            self.bands = ["u", "g", "r", "i", "z", "y"]
+        else:
+            self.bands = bands
+        self.data_path = None
+        self.transmission_tables = {}
+        self.normalized_system_response_tables = {}
+
+    def _get_data_path(self):
+        """Find/create a data directory for the bandpass data, located in same dir as passbands.py.
+
+        Note
+        ----
+        This is a temporary solution to be replaced when we implement Pooch throughout the project.
+        """
+        data_dir = os.path.join(Path(__file__).parent, "band_data")
+        if not os.path.exists(data_dir):
+            os.makedirs(data_dir)
+        self.data_path = data_dir
+
+    def _download_transmission_table(self, band_name: str, output_file: str, timeout: int = 5) -> bool:
+        """Download a transmission table for a single band and save it to a file.
```

Comment:
I would rename this to `_download_lsst_transmission_table` since this is very LSST-specific, or making this more general (for other filters). Also I think for LSST filters the best place to download (with the most up-to-date info) may be somewhere in the lsst-stack/Rubin site? (I can look this up)

Reply:
Linking this to #55 and will address this there, as I expect to scoop out a large amount of the current download functionality for that